src/openfermion/measurements/_probability_distributions.py

```diff
@@ -0,0 +1,363 @@
+                for j in range(num_vectors - 1)]):
+            raise ValueError('''Starting amplitudes must be different.''')
+
+        # The amplitude distributions are not being updated at this point
+        # (instead the data for the update is stored in p_vecs). As such,
+        # we just store this as a prior.
+        self._prior_amplitude_mean = amplitude_mean
+        self._prior_amplitude_var = amplitude_var
+
+        # The current estimate of the amplitudes is just the mean
+        # of the given guess.
+        self._amplitude_estimates = numpy.array(amplitude_mean)
+
+        if fourier_vector is not None:
+            self._fourier_vectors = numpy.array(fourier_vector)
+        else:
+            # Initialize probability distributions
+            self._fourier_vectors = numpy.zeros((self._num_freqs*2+1,
+                                                 self._num_vectors))
+            for j in range(self._num_vectors):
+                self._fourier_vectors[0, j] = 1
+
+        # This stores data from repeated Bayesian updates in order to
+        # calculate data of the updated distribution as required.
+        self._p_vecs = []
+
+        self._inv_covar_mat = numpy.linalg.inv(
+            numpy.array(self._prior_amplitude_var))
+
+    def get_real_dist(self, x_vec=None):
+        """
+        Generates the real distribution over a sequence of points
+
+        Args:
+            x_vec (numpy array or list): the points at which the distribution is generated.
+                If None, defaults to 101 points between -pi and pi.
+        Returns:
+            y_vecs (numpy array): the value of each distribution at the given x points.
+                y_vecs[j,k] = f_j(x_vec[k])
+        """
+        # Set default x vector
+        if x_vec is None:
+            x_vec = numpy.linspace(-pi, pi, 101)
+
+        # Init storage to return
+        y_vecs = []
+
+        for j in range(self._num_vectors):
+
+            # Get vector and evaluate corresponding function
+            # at the chosen set of points
+            vector = self._fourier_vectors[:, j]
+            y_vec = [self._make_dist(vector, theta) for theta in x_vec]
+
+            # Store and plot data
+            y_vecs.append(y_vec)
+
+        return numpy.array(y_vecs)
+
+    def _holevo_centers(self, vectors=None):
+        """
+        Gets the Holevo average phase from a distribution.
+        Args:
+            vectors (numpy array): vectors to calculate the average of. If None,
+                defaults to the vectors stored in the class.
+        Returns:
+            centers (numpy array): the corresponding average angle of the
+                distribution.
+        """
+        if vectors is None:
+            vectors = self._fourier_vectors
+
+        centers = numpy.angle(vectors[2, :] + 1j*vectors[1, :])
+        return centers
+
+    def _holevo_variances(self, vectors=None, maxvar=4*numpy.pi**2/12):
```

Comment:
This hidden method isn't used anywhere in the class.

Reply:
Same as above.